kernel/src/process/posix_thread/futex.rs

```diff
@@ -41,7 +41,7 @@ pub fn futex_wait_bitset(
     // lock futex bucket ref here to avoid data race
     let mut futex_bucket = futex_bucket_ref.lock();
 
-    if futex_key.load_val() != futex_val {
+    if futex_key.load_val()? != futex_val {
```

Comment:
I use your test program from #1334, and Linux program reports `EAGAIN`. If you use `?` here, Asterinas will report `EFAULT`. Maybe you can try the following form
```rust
if ! futex_key.load_val().is_ok_and(|val| val == futex_val) {
    return_errno_with_message!(Errno::EAGAIN, "futex value cannot be accessed");
}
```

Reply:
Modified🫡